pkg/prompts/prompts.go

```diff
@@ -31,10 +31,20 @@ const (
 	MoreInfo = "More Info"
 	Done     = "Done"
 	Cancel   = "Cancel"
+
+	LessThanEq = "Less Than Or Eq"
+
+	MoreThanEq = "More Than Or Eq"
+
+	MoreThan = "More Than"
 )
 
 var errNoKeys = errors.New("no keys")
 
+type Comparator struct {
```

Comment:
I mean instead of 
```
                               labelValue := compareValue.CompareValue
                                switch compareValue.CompareType {
                                case LessThanEq:
                                        if val > compareValue.CompareValue {
                                                return fmt.Errorf(fmt.Sprintf("the value must be smaller than or equal to %s (%d)", compareLabel, labelValue))
                                        }
                                case MoreThan:
                                        if val <= compareValue.CompareValue {
                                                return fmt.Errorf(fmt.Sprintf("the value must be bigger than %s (%d)", compareLabel, labelValue))
                                        }
                                case MoreThanEq:
                                        if val < compareValue.CompareValue {
                                                return fmt.Errorf(fmt.Sprintf("the value must be bigger than or equal to %s (%d)", compareLabel, labelValue))
                                        }
                                }
```

maybe you can use `if err := compareValue.validate(val) { return err }` 
with a suitable validate func defined on CompareType